### PR TITLE
Added module win_domainjoin.

### PR DESCRIPTION
**Pull request by trondhindenes**
win_domainjoin allows Ansible to join a windows computer to a domain. Parameters include the ability to auto-restart after a successfull join (this is turned off by default but should possily be enabled) and which OU path (by DistinguishedName) to put the computer into.

### CODE REVIEW

File: library/windows/win_domainjoin [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    required: true
+    default: null
+    aliases: []
+  domainpassword:
+    description:
+      - password for the domainusername account    
+	required: true
+    default: null
+    aliases: []
+  OUPath:
```
Comment (cchurch) [inline library/windows/win_domainjoin:50]: I would lean toward this parameter being lower case to be consistent with the others (and Ansible modules in general); just my 2 cents.

File: library/windows/win_domainjoin [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    description:
+      - The OU in which to place the computer
+    default: null
+    aliases: []
+  Restart:
```
Comment (cchurch) [inline library/windows/win_domainjoin:55]: Parameter name is capitalized here, but not when used below.  I know PowerShell variables aren't case-sensitive, but code and docs should at least match.

File: library/windows/win_domainjoin.ps1 [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+$result = New-Object psobject @{
+    changed = $false
+}
+
+If ($params.domainname) {
+    $domainname = $params.domainname
+}
+Else {
+    Fail-Json $result "mising required argument: domainname"
```
Comment (cchurch) [inline library/windows/win_domainjoin.ps1:33]: s/mising/missing

File: library/windows/win_domainjoin.ps1 [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+{
+    $JoinDomain = $true
+
+}
+
+
+if ($JoinDomain)
+{
+    #Construct credobject from username/password
+    #Test parameters
```
Comment (cchurch) [inline library/windows/win_domainjoin.ps1:79]: I don't see any need to keep your test lines around...

File: library/windows/win_domainjoin.ps1 [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            $ErrorActionPreference = "Stop"
+            Add-Computer -DomainName $domainname -Credential $credential -OUPath $OUPath -WarningVariable WarningVar -restart:$restart
+        }
+        catch
+        {
+            Fail-Json $result $_.Exception.Message
+        }
+    }
+
+    Set-Attr $result "domainjoin_result" $success
```
Comment (cchurch) [inline library/windows/win_domainjoin.ps1:114]: Where is $success defined?

File: library/windows/win_domainjoin.ps1 [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    $result.changed = $true
+}
+Else
```
Comment (cchurch) [inline library/windows/win_domainjoin.ps1:117]: This Else block is not really necessary since you already set $result.change = $false above.

**Comment by trondhindenes** [comment]
Lots of good comments here Chris, thanks. I'm closing this PR and will open a new one with revised code.

**Pull request closed without merging**